front: restructre and optimize test list

Maymanaf opened the pull request:
closes #11442

Maymanaf committed:
front: reduce unnecessary timeouts

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

SharglutDev commented on front/tests/pages/operational-studies/operational-studies-page.ts:
```diff
@@ -89,28 +89,20 @@ class OperationalStudiesPage extends CommonPage {
 
   // Click on the button to add a scenario train.
-  async clickOnAddTrainButton() {
+  async openTrainScheduleForm() {
```
Since we do not handle train schedule only
```suggestion
  async openTimetableItemForm() {
```

SharglutDev commented on front/tests/pages/operational-studies/scenario-page.ts:
```diff
@@ -117,35 +117,31 @@ class ScenarioPage extends CommonPage {
-  async openScenarioByTestId(scenarioTestId: string) {
-    await this.page.getByTestId(scenarioTestId).first().hover({ trial: true });
-    await this.page.getByTestId(scenarioTestId).getByTestId('openScenario').click();
+  // Open a Scenario by its name (The Test ID is the same as the Name).
```
```suggestion
  // Open a scenario by its name
```

SharglutDev commented on front/tests/pages/rolling-stock/rolling-stock-selector.ts:
```diff
@@ -73,29 +70,44 @@ class RollingStockSelector extends CommonPage {
 
-  getRollingstockCardByTestID(rollingstockTestID: string) {
-    return this.rollingStockSelectorModal.getByTestId(rollingstockTestID);
+  getRollingstockCardByName(rollingstockName: string) {
```
Should this be `private` ?

Maymanaf replied:
it's used in test 017

SharglutDev commented on front/tests/008-op-times-and-stops-tab.spec.ts:
```diff
@@ -201,13 +204,13 @@ test.describe('Times and Stops Tab Verification', () => {
     // Delete a row and verify changes
     await timesAndStopsTab.verifyDeleteButtons(2);
-    await timesAndStopsTab.deleteButtons.nth(0).click();
+    await timesAndStopsTab.deleteRow(0);
```
We don't really delete the row, we just clear it

Yohh commented on front/tests/pages/stdcm/stdcm-page.ts:
```diff
@@ -132,7 +131,7 @@ class STDCMPage extends HomePage {
   async verifyValidSimulationLaunch(): Promise<void> {
     await this.launchSimulation();
-    await this.simulationStatus.waitFor({ timeout: STDCM_SIMULATION_TIMEOUT });
+    await this.simulationStatus.waitFor();
```
no state to wait for?

Maymanaf replied:
The default state is` visible`

Yohh commented on front/tests/006-op-rolling-stock-tab.spec.ts:
```diff
@@ -50,51 +50,40 @@ test.describe('Rolling stock Tab Verification', () => {
     const comfortACRadioText = await rollingStockSelector.comfortACButton.innerText();
-    await rollingStockSelector.comfortACButton.click();
-    await rollingstockCard.locator('button').click();
+    await rollingStockSelector.selectRollingStockCard(dualModeRollingStockName, true, true);
```
can you change the parameters of `selectRollingStockCard()` with an object?

having two consecutive booleans is not explicit, an object would make it far more readable

Yohh commented on front/tests/015-train-timetable.spec.ts:
```diff
@@ -103,71 +103,7 @@ test.describe('Verify train schedule elements and filters', () => {
     await scenarioTimetableSection.verifyEachTrainSimulation();
   });
 
-  // TODO Paced train - remove this test in https://github.com/OpenRailAssociation/osrd/issues/10791
```
should be done in its own commit

Yohh commented on front/tests/015-train-timetable.spec.ts:
```diff
@@ -268,4 +204,68 @@ test.describe('Verify train schedule elements and filters', () => {
     }
   });
+
+  // TODO Paced train - remove this test in https://github.com/OpenRailAssociation/osrd/issues/10791
```
same as above

Yohh commented on front/tests/pages/operational-studies/paced-train-section.ts:
```diff
@@ -52,14 +50,13 @@ class PacedTrainSection extends CommonPage {
 
   // Only the zone with the role button opens the occurrence list
-  async getPacedTrainToClickableZone(index: number) {
+  private async getPacedTrainToClickableZone(index: number) {
```
this and next like this have nothing related to restructuration, should be done in its own commit

Yohh reviewed (commented):
left a few comments

too much changes, related to too much different purposes are done in a single commit, restructuring is not renaming, not changing some logic here and there... 

can you make more commits and add some description? it will be better for the git history and more comfortable for the reviewers.

Maymanaf committed:
front: rename and scope multiple functions as private for e2e tests

Renamed several functions in the page object for better clarity and consistency, and removed unnecessary comments. Also updated their visibility to `private` to better encapsulate internal logic.

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

Maymanaf committed:
front: rename and restructure the e2e test list

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

Maymanaf committed:
front: refactor RollingStockSelector page object for clarity and encapsulation

Renamed and restructured several methods for better clarity and type safety. Introduced a parameter object in 'selectRollingStockCard' to avoid positional booleans. Added helper methods to verify selected comfort and inactive states.

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

Maymanaf committed:
front: handle empty simulation result to avoid timeout

Signed-off-by: maymanaf <med.aymen.naf@gmail.com>

Yohh commented on front/tests/pages/operational-studies/operational-studies-page.ts:
```diff
@@ -89,28 +89,20 @@ class OperationalStudiesPage extends CommonPage {
 
-  // Open Route Tab
-  async clickOnRouteTab() {
+  async selectRouteTab() {
```
nit: for those `select..`, what about `toggle...` since this is not a selector? (just a suggestion, feel free to avoid it)

Math-R replied:
or whatabout `click...` as it wait for a `.click()` and that's all ?

Maymanaf replied:
what about  `openRouteTab()` ?

Yohh replied:
ok go for it, for this `select...` and the others below

Math-R reviewed (approved):
Looks good to me, thanks for the separation of commit, it mades it far more easier to review
